From 8678f63f4c9cbcaac14e4a2334957cc060f81205 Mon Sep 17 00:00:00 2001
From: mssurajkaiga <mssurajkaiga@gmail.com>
Date: Fri, 3 May 2013 06:02:11 +0530
Subject: [PATCH] Fixed bug 340094

---
 libnautilus-private/nautilus-canvas-item.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libnautilus-private/nautilus-canvas-item.c b/libnautilus-private/nautilus-canvas-item.c
index 9747349..8a93524 100644
--- a/libnautilus-private/nautilus-canvas-item.c
+++ b/libnautilus-private/nautilus-canvas-item.c
@@ -1973,6 +1973,8 @@ nautilus_canvas_item_get_max_text_width (NautilusCanvasItem *item)
 
 	canvas_item = EEL_CANVAS_ITEM (item);
 
+	if ((item->details->icon_rect.x1 - item->details->icon_rect.x0) > MAX_TEXT_WIDTH_STANDARD)
+		return 	(item->details->icon_rect.x1 - item->details->icon_rect.x0) * canvas_item->canvas->pixels_per_unit;
 	return MAX_TEXT_WIDTH_STANDARD * canvas_item->canvas->pixels_per_unit;
 }
 
-- 
1.8.1.2

